packages/core/src/modules/credentials/protocol/v2/handlers/V2OfferCredentialHandler.ts

```diff
@@ -0,0 +1,118 @@
+import type { CredentialExchangeRecord } from '../../../repository/CredentialExchangeRecord'
+import type { V2CredentialService } from '../V2CredentialService'
+
+import { AriesFrameworkError } from '../../../../../../src/error/AriesFrameworkError'
+import { createOutboundMessage, createOutboundServiceMessage } from '../../../../../agent/helpers'
+import { ServiceDecorator } from '../../../../../decorators/service/ServiceDecorator'
+import { DidCommMessageRole } from '../../../../../storage'
+import { V2OfferCredentialMessage } from '../messages/V2OfferCredentialMessage'
+import { V2ProposeCredentialMessage } from '../messages/V2ProposeCredentialMessage'
+
+export class V2OfferCredentialHandler implements Handler {
+  private credentialService: V2CredentialService
+  private agentConfig: AgentConfig
+  private mediationRecipientService: MediationRecipientService
+  public supportedMessages = [V2OfferCredentialMessage]
+
+  private didCommMessageRepository: DidCommMessageRepository
+
+  public constructor(
+    credentialService: V2CredentialService,
+    agentConfig: AgentConfig,
+    mediationRecipientService: MediationRecipientService,
+    didCommMessageRepository: DidCommMessageRepository
+  ) {
+    this.credentialService = credentialService
+    this.agentConfig = agentConfig
+    this.mediationRecipientService = mediationRecipientService
+    this.didCommMessageRepository = didCommMessageRepository
+  }
+
+  public async handle(messageContext: InboundMessageContext<V2OfferCredentialMessage>) {
+    const credentialRecord = await this.credentialService.processOffer(messageContext)
+
+    const offerMessage = await this.didCommMessageRepository.findAgentMessage({
+      associatedRecordId: credentialRecord.id,
+      messageClass: V2OfferCredentialMessage,
+    })
+
+    const proposeMessage = await this.didCommMessageRepository.findAgentMessage({
+      associatedRecordId: credentialRecord.id,
+      messageClass: V2ProposeCredentialMessage,
+    })
+
+    if (!offerMessage) {
+      throw new AriesFrameworkError('Missing offer message in V2OfferCredentialHandler')
+    }
+
+    const shouldAutoRespond = this.credentialService.shouldAutoRespondToOffer(
+      credentialRecord,
+      offerMessage,
+      proposeMessage ?? undefined
+    )
+
+    if (shouldAutoRespond) {
+      return await this.createRequest(credentialRecord, messageContext, offerMessage)
+    }
+  }
+
+  private async createRequest(
+    record: CredentialExchangeRecord,
+    messageContext: HandlerInboundMessage<V2OfferCredentialHandler>,
+    offerMessage?: V2OfferCredentialMessage
+  ) {
+    this.agentConfig.logger.info(
+      `Automatically sending request with autoAccept on ${this.agentConfig.autoAcceptCredentials}`
+    )
+
+    if (messageContext.connection) {
+      const { message, credentialRecord } = await this.credentialService.createRequest(
+        record,
+        {},
+        messageContext.connection.did
+      )
```

Comment:
As mentioned before, I think ti would look cleaner to add the did to the options object:

```suggestion
      const { message, credentialRecord } = await this.credentialService.createRequest(
        record,
        {
          holderDid: messageContext.connection.did
        } 
      )
```

Reply:
I could have sworn we decided not to put it in the options object as it is Indy specific, so from a previous review the decision was made to make it (temporarily) a parameter while we decided how to address this issue.

That said, I have made the change anyway.